testing/internal/e2e/tests/aws/session_recordings_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package aws_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliSessionRecordingsDownload validates that an admin can download session recording files from AWS S3 Bucket.
+// It creates a Storage Bucket and sets it to a Target with enabled session recording.
+// Then it connects to the Target and executes basic commands.
+// Then it downloads created session recording file from the AWS S3 Bucket via boundary CLI and verifies its content.
+func TestCliSessionRecordingsDownload(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+
+	// Create Org and Project
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId)
+
+	// Create storage bucket
+	newBucketId := boundary.CreateNewStorageBucket(t, ctx, newOrgId, c.AwsBucketName)
+
+	// Create static credentials
+	newCredentialStoreId := boundary.CreateNewCredentialStoreStaticCli(t, ctx, newProjectId)
+	pwCredentialsId := boundary.CreateNewStaticCredentialPasswordCli(t, ctx, newCredentialStoreId, c.TargetSshUser, "password")
+
+	// Create SSH target with storage bucket
+	newTargetId := boundary.CreateNewSshTargetWithStorageBucketCli(t, ctx, newProjectId, c.TargetPort, c.TargetIp, newBucketId)
+	boundary.AddInjectedCredentialSourceToTargetCli(t, ctx, newTargetId, pwCredentialsId)
+
+	// Connect to target and echo test value.
+	testValue := "Session Recording Test"
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"connect",
+			"-target-id", newTargetId,
+			"-exec", "/usr/bin/ssh", "--",
+			"-o", "UserKnownHostsFile=/dev/null",
+			"-o", "StrictHostKeyChecking=no",
+			"-p", "{{boundary.port}}",
+			"{{boundary.ip}}",
+			fmt.Sprintf("echo %s", testValue),
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Download session recording file from AWS S3 once it is available
```

Comment:
we should validate that the session recording state is set to finished/completed.